src/lib/caching/structures/base/Structure.ts

```diff
@@ -1,23 +1,22 @@
-import type { Client } from '../../../Client';
-import type { WebhookClient } from '../../../WebhookClient';
+import type { Client } from '../../../client/Client';
 
 /**
  * The base class for Structures
  */
-export abstract class Structure {
+export abstract class Structure<T = Client> {
 
 	/**
 	 * The id to be defined in Structures
 	 */
 	public abstract readonly id: string;
 
 	// eslint-disable-next-line no-useless-constructor
-	public constructor(public readonly client: Client | WebhookClient) { }
+	public constructor(public readonly client: T) { }
 
 	/**
 	 * Basic clone method
 	 */
-	public clone<T = Structure>(): T {
+	public clone<V = Structure>(): V {
```

Comment:
Shouldn't this return a "this" type?

Reply:
It's not the same instance though? While I didn't originally write this, only took the T... this implies the return of that method would === this, instead of only deeply equal it.

Reply:
Fair, but does `clone` right now inherit the type from the structure you clone or does it lose it?